src/java/org/apache/cassandra/locator/EndpointsByRange.java

```diff
@@ -54,9 +55,13 @@ public void putAll(Range<Token> range, EndpointsForRange replicas, Conflict igno
             get(range).addAll(replicas, ignoreConflicts);
         }
 
-        public EndpointsByRange asImmutableView()
+        public EndpointsByRange build()
         {
-            return new EndpointsByRange(Collections.unmodifiableMap(Maps.transformValues(map, EndpointsForRange.Mutable::asImmutableView)));
+            Map<Range<Token>, EndpointsForRange> map =
+                    Collections.unmodifiableMap(
+                            new HashMap<>(
```

Comment:
Is it really cheaper than, say `ImmutableMap.copyOf()`?

Reply:
No, it is not.  Good point - thanks!